nixos/modules/hardware/cpu/amd-ryzen-smu.nix

```diff
@@ -0,0 +1,29 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let
+  cfg = config.hardware.cpu.amd.ryzen_smu;
+  ryzen_smu = config.boot.kernelPackages.ryzen_smu;
+  ryzenadjCheck = lib.warnIf (builtins.elem pkgs.ryzenadj config.environment.systemPackages)
+    "ryzenadj is in environment.systemPackages, try hardware.cpu.amd.ryzen-smu and programs.ryzen_monitor_ng for more features.";
```

Comment:
This doesn't seem to work at all, it seems either I don't understand Nix thoroughly, or this just doesn't make sense. This is my fault please forgive me.

Reply:
My guess is that it isn't getting used in the function, so lazy evaluation means it never runs. 

I am going to take it out though, because a wiki listing both ryzenadj and ryzen_smu would be more useful.